### ISSUE TITLE
Joystick.cpp issue on Haxe 3.4.1

### ISSUE THREAD

**Issue by starburst997** [openfl/lime#943]
On Haxe 3.4.1:
```
Error: Joystick.cpp
./src/lime/ui/Joystick.cpp(75) : error C2668: 'String::String' : ambiguous call to overloaded function
        c:\haxetoolkit\haxe\lib\hxcpp\3,4,64\include\hxString.h(104): could be 'String::String(const Dynamic &)'
        c:\haxetoolkit\haxe\lib\hxcpp\3,4,64\include\hxString.h(73): or       'String::String(const cpp::Variant &)'
        while trying to match the argument list '(hx::Object *)'
./src/lime/ui/Joystick.cpp(83) : error C2668: 'String::String' : ambiguous call to overloaded function
        c:\haxetoolkit\haxe\lib\hxcpp\3,4,64\include\hxString.h(104): could be 'String::String(const Dynamic &)'
        c:\haxetoolkit\haxe\lib\hxcpp\3,4,64\include\hxString.h(73): or       'String::String(const cpp::Variant &)'
        while trying to match the argument list '(hx::Object *)'
```

Did not dig any further, just installed back 3.4.0 and it was working. (latest version as of now of hxcpp + lime + openfl)

**Comment by Gama11** [comment]
See here:

- https://github.com/openfl/openfl/issues/1531
- https://github.com/HaxeFoundation/haxe/issues/6103

Already fixed on Haxe dev, and Haxe 3.4.2 should be released soon.

**Comment by starburst997** [comment]
Ahhh cool!
